dipy/align/api.py

```diff
@@ -0,0 +1,854 @@
+"""
+
+Registration API: simplified API for registration of MRI data and of
+streamlines.
+
+
+"""
+import collections
+import numbers
+import numpy as np
+import nibabel as nib
+from dipy.align.metrics import CCMetric, EMMetric, SSDMetric
+from dipy.align.imwarp import (SymmetricDiffeomorphicRegistration,
+                               DiffeomorphicMap)
+
+from dipy.align.imaffine import (transform_centers_of_mass,
+                                 AffineMap,
+                                 MutualInformationMetric,
+                                 AffineRegistration)
+
+from dipy.align.transforms import (TranslationTransform3D,
+                                   RigidTransform3D,
+                                   AffineTransform3D)
+
+import dipy.core.gradients as dpg
+import dipy.data as dpd
+from dipy.align.streamlinear import StreamlineLinearRegistration
+from dipy.tracking.streamline import set_number_of_points
+from dipy.tracking.utils import transform_tracking_output
+from dipy.io.streamline import load_trk
+
+
+syn_metric_dict = {'CC': CCMetric,
+                   'EM': EMMetric,
+                   'SSD': SSDMetric}
+
+
+def _input_as_img_arr_or_path(data, affine=None):
```

Comment:
I do not think this function should be in this module. it is really useful and can be used everywhere. 

At the same time, I feel this is the role of the workflows to do and we should not start to mix it and be rigorous with this point.

So I have this double feeling and we need to make a decision about this point.

Reply:
I'd be happy to move it elsewhere. In `dipy.io` somewhere?

Reply:
good idea, `dipy.io.utils` seems a good place!

Reply:
Done in 25f24dc8b49f0fbd2405